sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala

```diff
@@ -36,6 +38,15 @@ class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with T
 
   import ScriptTransformationIOSchema._
 
+  protected override def getTestResourcePath(fileName: String): String = {
```

Comment:
What about we just define a method such as:

```scala
def withScriptFile(script: String)(f: File => Unit) = {
  val path = Utils.createTempDir()
  path.delete()
  Files.write(path, script.getBytes(StandardCharsets.UTF_8))
  try f(path) finally Utils.deleteRecursively(path)
}
```

at `BaseScriptTransformationSuite`, and call like:

```scala
withScriptFile(
  """
    |import sys
    |for line in sys.stdin:
    |    (a, b, c, d, e) = line.split('\t')
    |    sys.stdout.write('\t'.join([a, b, c, d, e]))
    |    sys.stdout.flush()
  """.stripMargin) { file =>
    // test with `file`.
}
```

Reply:
Since same script, if change so..too many duplicated file message